### PR TITLE
Block Editor Handbook: Improved the readability and phrasing and corrected mistakes

### PR DESCRIPTION
**Pull request by ghost**
Corrected weird and awkward phrasings and sentences, and made some things more specific.

Some examples and thoughts behind the changes:
Why would the Block Editor documentation link to the Block Editor documentation. I'm already there right? We're linking to the How-To guides.

Or why do we write Block Inserter capitalized, but not Block Editor?

Sometimes the order of words was just plain weird. "you will find here guides to help you"

There was a quick link, just before the heading Quick Links, which didn't make sense to me. Also, it's one of the first links in the documentation and it links away from the documentation to a support article so that asks for a bit more context. Clearly stating you're going away from dev focussed content to end-user focussed content.

### CODE REVIEW

**Comment by alexstine** [comment]
The whole case issue around Gutenberg is really annoying. I have seen this problem too. Some is an uppercase letter for each word and some is sentence format.

**Comment by ghost** [comment]
> The whole case issue around Gutenberg is really annoying.

I agree, wonder how this is handled in WordPress docs. Might need a global document where we define what gets capitalized and what doesn't. I also see Block and block everywhere, same goes for Editor and editor.

File: docs/README.md [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
+**Gutenberg** is a codename for a whole new paradigm in WordPress site building and publishing, that aims to revolutionize the entire publishing experience as much as Gutenberg did the printed word. Right now, the project is in the second phase of a four-phase process that will touch every piece of WordPress -- Editing, **Customization** (which includes Full Site Editing, Block Patterns, Block Directory and Block based themes), Collaboration, and Multilingual -- and is focused on a new editing experience, the block editor (which is the topic of the current documentation).
 
 ![Quick view of the block editor](https://raw.githubusercontent.com/WordPress/gutenberg/trunk/docs/assets/quick-view-of-the-block-editor.png)
 
-**Legend :**
+**Legend:**
 
 1. Block Inserter
-2. Block editor content area
+2. Block Editor Content Area
```
Comment (ryanwelcher) [inline docs/README.md]: Just want to confirm that Block Editor ( capital E ) is how it is spelled in other places

Comment (ryanwelcher) [reply docs/README.md]: Let's change all of these to sentence case:

1. Block inserter
2. Block editor content area
3. Settings sidebar

**Review by ryanwelcher — CHANGES_REQUESTED** [review]
Just a small change then I think we're good to go!

**Review by ryanwelcher — APPROVED** [review]
LGTM!

**Review by aristath — APPROVED** [review]
LGTM!
There was a failing test preventing this from getting merged but it's not related this PR, so I restarted the test and once it passes we can merge this 👍

**Pull request merged**